ergonomics_profiles/readme.md

```diff
@@ -0,0 +1,193 @@
+
+**_GC selection_**
+
+| Memory     | Processors | GC selected |
+| ---------- | ---------- | ----------- |
+| \>=1792 MB | \>1        | G1          |
+| Otherwise  |            | Serial      |
+
+_**GC threads**_
+
+| GC        | \# of CPUs | ConcGCThreads                   | ParallelGCThreads      |
+| --------- | ---------- | ------------------------------- | ---------------------- |
+| Serial GC | any        | 0                               | 0                      |
+| G1 GC     | 1-8        | max((ParallelGCThreads+2)/4, 1) | ceil(#CPUs)            |
+| G1 GC     | \>8        | max((ParallelGCThreads+2)/4, 1) | 8 + (#CPUs-8) \* (5/8) |
+
+Which GC is used by the application will impact the amount of native memory consumed by the JVM, as well as impact throughput and latency.
+
+**CPU allocation**
+
+The CPU allocation, in most systems, is about CPU time instead of the CPU count. This is especially true in container-based environments. The JVM will expose the number of _active processors_ through the API `Runtime.getRuntime().availableProcessors()`, which in turn is used by the JVM itself to size internal thread pools and then read by 3rd-party frameworks and libraries to size their thread pools.
+
+In systems where CPU time is managed with limits, the JVM will often be at risk of suffering from _CPU throttling_, and it must be careful not to exaggerate the number of threads it creates. Otherwise, the application will be impacted even further.
+
+CPU allocation is the most challenging aspect of ergonomics, as the consequences are unpredictable without knowing what workload the JVM is running and what it will actually need. Currently, the JVM calculates the number of active processors with the following check:
+
+* Server: 1 “active processor” for each CPU
+* Cgroups-based counting on cpu_quota/cpu_period
+
+The user may override the calculation with the parameter `-XX:ActiveProcessorCount=<n>`. In practice, developers tend not to refrain from using this and will expect the JVM to size itself accordingly. But a JVM with a single observed "active processor" may still run multiple threads in parallel. Although they will consume the allowed CPU time much faster, this is negligible in some applications, as the application may have significant wait times for IO operations to finish.
+
+Description
+-----------
+
+We propose adding the concept of Ergonomics Profiles while naming the existing default ergonomics as `shared`, and adding a second profile called `dedicated` for when the JVM is to run under environments with resources dedicated to it. New profiles may be added in the future.
+
+**_Selecting a profile_**
+
+To select a profile, we propose a new flag:
+
+    -XX:ErgonomicsProfile=<shared|dedicated>
+
+Users may also select a profile by setting this flag in the environment variable `JAVA_TOOL_OPTIONS`.
```

Comment:
Is there a reason to call out this env var and not others like `JDK_JAVA_OPTIONS`?